keps/sig-apps/4563-eviction-request-api/README.md

```diff
@@ -0,0 +1,1948 @@
+   request and should wait for either the pod deletion or removal of the evacuation request. The
+   owning controller of the pod should observe the pod's conditions and respond to the
+   `EvacuationRequested` by accepting it and setting an `EvacuationInitiated` condition to `True` in
+   the pod conditions.
+2. `EvacuationInitiated` condition should be set by the owning controller to signal that work is
+   being done to either remove or evacuate/migrate the pod to another node. The draining
+   process/controller should wait a reasonable amount of time (3 minutes) to observe the appearance
+   of the condition or change of the condition status to `True`. The draining process should then
+   skip such a pod and leave its management to the owning controller. If `EvacuationInitiated`
+   condition does not appear after 3 minutes, the draining process will begin evicting or deleting
+   the pod. If the owning controller is unable to remove or migrate the pod, it should set the
+   `EvacuationInitiated` condition status back to `False` to give the eviction a chance to start.
+
+```golang
+
+type PodConditionType string
+
+const (
+	...
+	EvacuationRequested PodConditionType = "EvacuationRequested"
+	EvacuationInitiated PodConditionType = "EvacuationInitiated"
+)
+
+const (
+	...
+	PodReasonNodeMaintenance = "NodeMaintenance"
+)
+
+```
+
+There are multiple issues with this approach:
+- Multiple eviction requesters can try to request an eviction of a pod at the same time and there is
+  no simple way to track this in a single condition. It is much easier to handle concurrency on a
+  standalone resource.
+- The observability of the eviction progress, interceptor reference and other useful information is
+  lacking as we can see requested in the [User Stories](#user-stories-optional).
+
+
+### Enhancing PodDisruptionBudgets
+
+The main function of today's PDBs is to count how many pods the application has via the selector.
+And how many of those pods are healthy and therefore how many can be disrupted.
+
+We would like to extend this API with the following functionality
+- Request the eviction of a pod.
+- Observe which pods are in the progress of being evicted.
+
+As a cluster admin it should be possible to request the eviction of any pod. Even the one that is
+guarded by a PDB. Currently, applications can set `maxUnavailable: 0` on a PDB and thus forbid any
+eviction. Among the reasons mentioned in the
+[EvictionRequest or Eviction subresource](#evictionrequest-or-eviction-subresource) section, it is
+also not possible to change the default eviction behavior for safety reasons. So an opt-in behavior
+would not be sufficient, as we need to solve this for all pods.
+
+We could track these requests in the PDB object itself. A PDB can track multiple pods with its
+selector. The pods do not have to be from the same application, even though they usually are. These
```

Comment:
>  The pods do not have to be from the same application, even though they usually are.

The original idea is that a single PDB is for a single application. Even though I can see how one might overuse it in the way you put it here, I'd prefer we don't in our official documents.